argilla/src/argilla/responses.py

```diff
@@ -183,6 +184,10 @@ def api_model(self):
         values = self.__responses_as_model_values(self.responses)
         for question_name, value in values.items():
             question = self._record.dataset.settings.questions[question_name]
+            if question is None:
+                raise ResponsesError(
+                    f"response invalid because targeted question with name={question_name} does not exist. Available questions are: {list(self._record.dataset.settings.questions._properties_by_name.keys())}"
```

Comment:
```suggestion
                    f"Record response is invalid because question with name={question_name} does not exist in the dataset ({self._record.dataset.name}). Available questions are: {list(self._record.dataset.settings.questions._properties_by_name.keys())}"
```